src/Microsoft.AspNetCore.Authentication/RemoteAuthenticationHandler.cs

```diff
@@ -69,7 +62,7 @@ public virtual Task<bool> ShouldHandleRequestAsync()
                 {
                     return true;
                 }
-                else if (authResult.Nothing)
+                else if (authResult.Skipped || authResult.Ignored)
```

Comment:
The explanation is here: https://github.com/aspnet/Security/pull/1266#discussion_r123054352.